src/gui/VerificationValidationWidget.cpp

```diff
@@ -190,22 +202,190 @@ void VerificationValidationWidget::dbPopulateDefaults() {
+}
+
+void VerificationValidationWidget::updateTestListWidget(QListWidgetItem* suite_clicked) {
+    QSqlQuery* q = new QSqlQuery(getDatabase());
+    q->prepare("Select testID from TestsInSuite Where testSuiteID = (SELECT id FROM TestSuites WHERE suiteName = :suiteName)");
+    q->bindValue(":suiteName", suite_clicked->text());
+    dbExec(q, !SHOW_ERROR_POPUP);
+
+    QListWidgetItem* item = 0;
+    while(q->next()){
+        int row = q->value(0).toInt() - 1;
+        item = testList->item(row);
+        if(suite_clicked->checkState()){
+            item->setCheckState(Qt::Checked);
+        } else {
+            item->setCheckState(Qt::Unchecked);
+        }
+        testListSelection(item);
+    }
+
+    if(!suite_clicked->checkState()){
+        suite_sa->item(0)->setCheckState(Qt::Unchecked);
+    }
+    checkSuiteSA();
+}
+
+void VerificationValidationWidget::testListSelection(QListWidgetItem* test_clicked) {
+    QSqlQuery* q1 = new QSqlQuery(getDatabase());
+    QSqlQuery* q2 = new QSqlQuery(getDatabase());
+    
+    q1->prepare("Select testSuiteID from TestsInSuite Where testID = (SELECT id FROM Tests WHERE testName = :testName)");
+    q1->bindValue(":testName", test_clicked->text());
+    dbExec(q1, !SHOW_ERROR_POPUP);
+    while(q1->next()){
+        QListWidgetItem* suite = suiteList->item(q1->value(0).toInt()-1);
+        if(!test_clicked->checkState()){
+            // If any test unchekced -> update test sa and suite containing test_clicked
+            if(suite->checkState()){
+                suite->setCheckState(Qt::Unchecked);
+                suite_sa->item(0)->setCheckState(Qt::Unchecked);
+            }
+        } else {
+            // Check if all test in a suite is checked  -> check suite
+            q2->prepare("Select testID from TestsInSuite Where testSuiteID = :suiteID");
+            q2->bindValue(":suiteID", q1->value(0).toInt());
+            dbExec(q2, !SHOW_ERROR_POPUP);
+            while(q2->next()){
+                QListWidgetItem* test = testList->item(q2->value(0).toInt()-1);
+                if(!test->checkState()){
+                    return;
+                }
+            }
+            suite->setCheckState(Qt::Checked);
+        }
+    }
+
+    if(!test_clicked->checkState()){
+        test_sa->item(0)->setCheckState(Qt::Unchecked);
+    }
+    checkSuiteSA();
+    checkTestSA();
+}
+
+void VerificationValidationWidget::userInputDialogUI(QListWidgetItem* test) {
+    QDialog* userInputDialog = new QDialog();
+    userInputDialog->setModal(true);
+    userInputDialog->setWindowTitle("Custom Inputs");
+
+    QVBoxLayout* vLayout = new QVBoxLayout();
+    QFormLayout* formLayout = new QFormLayout();
+
+    vLayout->addWidget(new QLabel(test->text()));
+
+    QLineEdit* input1 = new QLineEdit();
+    QLineEdit* input2 = new QLineEdit();
+    
+    formLayout->addRow(QString("Input 1:  "), input1);
+    formLayout->addRow(QString("Input 1:  "), input2);

```

Comment:
Think this was meant to be "Input 2: "

Reply:
Yeah, ignore that part. Not finalized yet.